packages/cli/cloud/src/link/action.ts

```diff
@@ -0,0 +1,204 @@
+import inquirer from 'inquirer';
+import chalk from 'chalk';
+
+import type { Answers } from 'inquirer';
+import type { CLIContext } from '../types';
+import type { CloudApiService } from '../services/cli-api';
+
+import { LocalSave } from '../services/strapi-info-save';
+import { cloudApiFactory, tokenServiceFactory, local } from '../services';
+import { promptLogin } from '../login/action';
+import { trackEvent } from '../utils/analytics';
+
+const QUIT_OPTION = 'Quit';
+
+interface LinkProjectValue {
+  name: string;
+  displayName: string;
+}
+
+interface LinkProjectAnswer extends Answers {
+  linkProject: LinkProjectValue;
+}
+
+interface LinkProjectInput extends Answers {
+  linkProject: LinkProjectValue | string;
+}
+
+type ProjectsList = {
+  name: string;
+  value: {
+    name: string;
+    displayName: string;
+  };
+}[];
+
+type Project = {
+  name: string;
+  displayName: string;
+  isMaintainer: boolean;
+};
+
+async function getExistingConfig(ctx: CLIContext, cloudApiService: CloudApiService) {
+  let existingConfig: LocalSave;
+  try {
+    existingConfig = await local.retrieve();
+    if (existingConfig.project) {
+      const { shouldRelink } = await inquirer.prompt([
+        {
+          type: 'confirm',
+          name: 'shouldRelink',
+          message: `A project named ${chalk.cyan(
+            existingConfig.project.displayName
+              ? existingConfig.project.displayName
+              : existingConfig.project.name
+          )} is already linked to this local folder. Do you want to update the link?`,
+          default: false,
+        },
+      ]);
+
+      if (!shouldRelink) {
+        await trackEvent(ctx, cloudApiService, 'didNotLinkProject', {
+          currentProjectName: existingConfig.project?.name,
+        });
+        return null;
+      }
+    }
+  } catch (e) {
+    ctx.logger.debug('Failed to get project config', e);
+    ctx.logger.error('An error occurred while retrieving config data from your local project.');
+    return null;
+  }
+
+  return existingConfig;
```

Comment:
I believe early return and const are preferred, but it's just an opinion.
```suggestion
  try {
    const existingConfig = await local.retrieve();
    if (existingConfig.project) {
      const { shouldRelink } = await inquirer.prompt([
        {
          type: 'confirm',
          name: 'shouldRelink',
          message: `A project named ${chalk.cyan(
            existingConfig.project.displayName
              ? existingConfig.project.displayName
              : existingConfig.project.name
          )} is already linked to this local folder. Do you want to update the link?`,
          default: false,
        },
      ]);

      if (!shouldRelink) {
        await trackEvent(ctx, cloudApiService, 'didNotLinkProject', {
          currentProjectName: existingConfig.project?.name,
        });
        return null;
      }
      return existingConfig;
    }
  } catch (e) {
    ctx.logger.debug('Failed to get project config', e);
    ctx.logger.error('An error occurred while retrieving config data from your local project.');
  }

  return null;
```